java/src/jmri/jmrix/loconet/LnPacketizer.java

```diff
@@ -485,11 +487,41 @@ public void startThreads() {
     Thread rcvThread;
     Thread xmtThread;
     
-  
     /**
-     * Flag that threads should terminate as soon as they can.
+     * Testing Only - request the rec & xmit threads to be terminated
+     * if set before start, one loop only is performed.
+     */
+    protected void setthreadStopRequestOn() {
+        threadStopRequest = true;
+    }
+
+    /**
+     * Testing Only - Spins its wheels for 100ms or threads terminated
+     * @return true threads were terminated
+     */
+     protected boolean waitForThreadsToStop() {
+        for (int i = 1; i < 10; i++) {
```

Comment:
Shouldn’t this method be calling join on the two threads involved instead of waiting for these flags to be set?

Not joining the thread leaves open the possibility that the threads will still be running after this loop completes.

Calling join makes the flags irrelevant.

Reply:
I don't think so.
A join will hang the current thread until that one dies, if it never dies it just hangs, we should handle it gracefully. Its just a test. In reality its a daemon and never gets killed deliberately. These stops are only for testing, no one should ever kill the thread in production, if that is going to happen they should not be daemons. 
But the first hurdle if find out why the tests were skipped. currently no threads are started during testing.